fearless/Modules/Staking/ValidatorInfo/ValidatorInfoViewModel/ValidatorInfoViewModelFactory.swift

```diff
@@ -0,0 +1,163 @@
+        var sections: [ValidatorInfoViewController.Section] = []
+
+        if let stakingViewModel = generateStakingViewModel(from: validatorInfo) {
+            sections.append((.staking, stakingViewModel))
+        }
+
+        if let identityViewModel = generateIdentityViewModel(from: validatorInfo) {
+            sections.append((.identity, identityViewModel))
+        }
+
+        return sections
+    }
+
+    func generateAccountViewModel(from validatorInfo: ValidatorInfoProtocol) -> ValidatorInfoAccountViewModelProtocol {
+        let userIcon = try? iconGenerator.generateFromAddress(validatorInfo.address)
+            .imageWithFillColor(.white,
+                                size: UIConstants.normalAddressIconSize,
+                                contentScale: UIScreen.main.scale)
+
+        let viewModel: ValidatorInfoAccountViewModelProtocol =
+            ValidatorInfoAccountViewModel(name: validatorInfo.identity?.displayName,
+                                          address: validatorInfo.address,
+                                          icon: userIcon)
+
+        return viewModel
+    }
+
+    // MARK: - Private functions
+    private func generateStakingViewModel(
+        from validatorInfo: ValidatorInfoProtocol) -> [ValidatorInfoViewController.Row]? {
+
+        guard let stakeInfo = validatorInfo.stakeInfo else { return nil }
+
+        var stakingViewModel: [ValidatorInfoViewController.Row] = []
+
+        let tokenFormatter = amountFormatterFactory.createTokenFormatter(for: asset)
+
+        let totalStakeLocalizedContent: LocalizableResource<TitleWithSubtitleViewModel> =
+            LocalizableResource { locale in
+
+                let title = R.string.localizable
+                    .stakingValidatorTotalStake(preferredLanguages: locale.rLanguages)
+
+                let subtitle = tokenFormatter.value(for: locale)
+                    .string(from: stakeInfo.totalStake) ?? ""
+
+                return TitleWithSubtitleViewModel(title: title,
+                                                  subtitle: subtitle)
+        }
+
+        let nominatorsLocalizedContent: LocalizableResource<TitleWithSubtitleViewModel> =
+            LocalizableResource { locale in
+                let title = R.string.localizable
+                    .stakingValidatorNominators(preferredLanguages: locale.rLanguages)
+                return TitleWithSubtitleViewModel(title: title, subtitle: String(stakeInfo.nominators.count))
+        }
+
+        let estimatedRewardLocalizedContent: LocalizableResource<TitleWithSubtitleViewModel> =
+            LocalizableResource { locale in
+                let percentageFormatter = NumberFormatter.percentAPY.localizableResource().value(for: locale)
+
+                let title = R.string.localizable
+                    .stakingValidatorEstimatedReward(preferredLanguages: locale.rLanguages)
+
+                let subtitle = percentageFormatter
```

Comment:
Looks like this one doesn't depend on locale so we can instantiate outside of the closure